frontend/src/routers/RoadmapRouter.tsx

```diff
@@ -91,9 +99,41 @@ const RoadmapRouterComponent = ({ userInfo }: { userInfo: UserInfo }) => {
 
   useEffect(() => {
     // Try to select roadmap given in route parameters
-    if (roadmapId) {
-      dispatch(roadmapsActions.selectCurrentRoadmap(+roadmapId));
+    if (urlRoadmapId) {
+      dispatch(roadmapsActions.selectCurrentRoadmap(+urlRoadmapId));
     }
+  }, [dispatch, urlRoadmapId]);
+
+  useEffect(() => {
+    // If selectedRoadmapId changes, update the url to match that
+    // (for cases where another browser tab switches roadmapId and the state gets synced into this tab)
+    let newPath: string;
+    const roadmapIdRegex = new RegExp('^roadmap/\\d+/');
+    if (!previousRoadmapId || selectedRoadmapId === previousRoadmapId) return;
+    if (selectedRoadmapId === undefined) {
+      newPath = '/overview';
+    } else if (
+      selectedRoadmapId !== previousRoadmapId &&
+      history.location.pathname.match(roadmapIdRegex)
+    ) {
+      newPath = history.location.pathname.replace(
+        roadmapIdRegex,
+        `roadmap/${selectedRoadmapId}/`,
```

Comment:
```suggestion
        `/roadmap/${selectedRoadmapId}/`,
```

Anchoring the regex requires the leading `/` here also

Reply:
Weird, it somehow still worked in my testing, I'll change it